src/main/java/hyperledger/besu/java/rest/client/service/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,237 @@
+        new TransactionResponseModel(transactionDetails, transferTransactionReceipt.toString());
+    return new ClientResponseModel(ErrorConstants.NO_ERROR, resultObject);
+  }
+
+  @Override
+  public ClientResponseModel read(String contractAddress, String functionName, String... params) {
+
+    // The flow should be
+    // 1. To create the payload for transaction
+    // 2. Ge the payload signed by the right key, this is available in wallet
+    // Note: The current code assumes certain parameters such as gasPrice
+    // Ref: https://docs.web3j.io/4.8.7/transactions/wallet_files/
+    Transaction transaction =
+        Transaction.builder()
+            .contractAddress(contractAddress)
+            .functionName(functionName)
+            .params(params)
+            .gasPrice(BigInteger.valueOf(ethConfig.getEthProperties().getGasPrice()))
+            .gasLimit(BigInteger.valueOf(ethConfig.getEthProperties().getGasLimit()))
+            .nonce(getNonce(credentials.getAddress()))
+            .build();
+    String response = readTransaction(transaction);
+    List<Type> uint = FunctionReturnDecoder.decode(response, transaction.getOutputParams());
+    log.info("responseValue: {} ", response);
+
+    // Store the result into a custom object variable
+    TransactionResponseModel resultObject = new TransactionResponseModel(uint);
+    return new ClientResponseModel(ErrorConstants.NO_ERROR, resultObject);
+  }
+
+  @Override
+  public ClientResponseModel deploy(final String compiledHexString) {
+
+    // Contract is deployed as a transaction as well
+    Transaction transaction =
+        Transaction.builder()
+            .compiledHexBinary(compiledHexString)
+            .gasPrice(BigInteger.valueOf(ethConfig.getEthProperties().getGasPrice()))
+            .gasLimit(BigInteger.valueOf(ethConfig.getEthProperties().getGasLimit()))
+            .nonce(getNonce(credentials.getAddress()))
+            .build();
+    EthSendTransaction ethSendTransaction = sendTransaction(transaction.getContractTransaction());
+
+    String transactionHash = ethSendTransaction.getTransactionHash();
+    log.info("transactionHash: {}", transactionHash);
+    log.info("result: {}", ethSendTransaction.getResult());
+
+    // add a blocking call logic here to wait for the transaction receipt
+    // for a given hash.
+    TransactionReceipt transferTransactionReceipt;
+    try {
+      transferTransactionReceipt =
+          transactionReceiptProcessor.waitForTransactionReceipt(transactionHash);
+    } catch (Exception e) {
+      throw new BesuTransactionException(
+          HYPERLEDGER_BESU_TRANSACTION_RECEIPT_ERROR, e.getMessage());
+    }
+    log.info("transaction found in block: {}", transferTransactionReceipt.getBlockNumber());
+
+    String contractAddress = transferTransactionReceipt.getContractAddress();
+    log.info("contract is deployed at: {}", contractAddress);
+    return new ClientResponseModel(ErrorConstants.NO_ERROR, contractAddress);
```

Comment:
How about returning transaction ID ?

Reply:
Thanks, sure, I will add test cases and identify the return payload.